service/controller/clusterapi/v29/network/cluster_persister.go

```diff
@@ -0,0 +1,74 @@
+package network
+
+import (
+	"context"
+	"encoding/json"
+	"net"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/cluster/v1alpha1"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/cluster-api/pkg/client/clientset_generated/clientset"
+)
+
+type ClusterPersisterConfig struct {
+	CMAClient clientset.Interface
+	Logger    micrologger.Logger
+}
+
+type ClusterPersister struct {
```

Comment:
WDYT about naming this `ClusterStatusPersister` - it's still a weird naming but it makes it a bit clearer for me where and what gets persisted.

Reply:
I don't like it but when this is important to you we can go with it.

Reply:
For me it gives very little information about what it does when reading code - maybe I need to get into the codebase more first but when reading it there were multiple questions in my head (What is cluster exactly, where does it persist and so on).

Reply:
We talked. We keep the names short. This here is WIP and when we put everything together in the `ipam` resource we take care of decent godoc for all the different implementations.